sky/jobs/utils.py

```diff
@@ -290,49 +304,125 @@ def cancel_job_by_name(job_name: str) -> str:
     return f'Job {job_name!r} is scheduled to be cancelled.'
 
 
-def stream_logs_by_id(job_id: int, follow: bool = True) -> str:
+def stream_managed_job_task_launch_logs(job_id: int,
+                                        task_id: int,
+                                        follow: bool = True) -> None:
+    """This method is specifically used for non-chain DAGs of managed jobs.
+    Each task in the managed job DAG is similar to a replica in a
+    serve application, and like the first part of `stream_replica_logs`, we
+    print the launch log of the task, which is stored in the controller and
+    redirected from sky.launch's stdout.
+    """
+    print(f'{colorama.Fore.YELLOW}Start streaming logs for launching process '
+          f'of task {task_id}.{colorama.Style.RESET_ALL}')
+
+    launch_log_file_name = get_launch_log_file_name(job_id, task_id)
+    if not launch_log_file_name.exists():
+        with ux_utils.print_exception_no_traceback():
+            raise ValueError(f'Launch log file {launch_log_file_name} '
+                             f'for task {task_id} not found.')
+
+    def finish_stream():
```

Comment:
```suggestion
    def finish_stream() -> bool:
```
pls double check other places in the code for return value type annotations.

Reply:
Thanks for the feedback! I think type annotations aren’t always needed, especially for a function like this that’s closer to a lambda where the return type is obvious and IDEs can infer it. See https://google.github.io/styleguide/pyguide.html#319-type-annotations, it’s more important for public APIs or stable interfaces.

Reply:
IIUC mypy won't check functions without return value annotated. Please double check.